lib/puma/minissl.rb

```diff
@@ -289,6 +295,36 @@ def check
           raise "Key not configured" if @key.nil? && @key_pem.nil?
           raise "Cert not configured" if @cert.nil? && @cert_pem.nil?
         end
+
+        # Controls session reuse.  Allowed values are as follows:
+        # * 'off' - matches the behavior of Puma 5.6 and earlier.  This is included
+        #   in case reuse 'on' is made the default in future Puma versions.
+        # * 'dflt' - sets session reuse on, with OpenSSL default cache size of
+        #   20k and default timeout of 300 seconds.
+        # * 's,t' - where s and t are integer strings, for size and timeout.
+        # * 's' - where s is an integer strings for size.
+        # * ',t' - where t is an integer strings for timeout.
+        # reuse can also be used as a bind query parameter with the above values.
+        #
+        def reuse=(reuse_str)
```

Comment:
I suggest a different API:

```ruby
def reuse=(reuse, size: nil, timeout: nil)
end
```

We'll do the work of parsing the connection string on the user's behalf in `dsl.rb`. 

Should we also have a separate DSL method for setting this?

Reply:
I'll look at it.

I thought using the same string for a query parameter and `ssl_bind` calls.  Part of that is based on the fact that SSL query strings keep getting longer and longer as we add more options.  And, we need them for restarts...

So, I thought combining size & timeout in one parameter was best?

Reply:
Confused.

> We'll do the work of parsing the connection string on the user's behalf in dsl.rb.

When one has an ssl bind, `Binder#parse` converts the bind query string to a hash via `Util.parse_query`.  Then the 'query' hash is passed to `MiniSSL::ContextBuilder.new` which initializes a `MiniSSL::Context`.  Hence, rather than dsl.rb, do you mean `MiniSSL::ContextBuilder.new`?

> Should we also have a separate DSL method for setting this?

DSL really has only one SSL method, `#ssl_bind`, which calls a convenience method `.ssl_bind_str`. At present, ssl options are passed to `#ssl_bind` as a hash, with the values essentially matching the query parameter string values.  It assembles the bind query string.

Reply:
@nateberkopec Still not quite sure what you're looking for here.  As we keep adding SSL options, the query string gets more 'not human readable'.  The code in this PR was an attempt to keep it so.  It's easy to use SSl's defaults, and people can change the setup if desired...

Reply:
> We'll do the work of parsing the connection string on the user's behalf in dsl.rb.
> Should we also have a separate DSL method for setting this?

dsl.rb has `ssl_bind`, which translates an options hash into a bind string.  So that's an odd place for it.

1. `def reuse=(reuse, size: nil, timeout: nil)` - it's somewhat odd to have a attribute setter accept multiple parameters?
2. The method should probably go in either [MiniSSL::ContextBuilder](https://msp-greg.github.io/puma/Puma/MiniSSL/ContextBuilder.html) or [MiniSSL::Context](https://msp-greg.github.io/puma/Puma/MiniSSL/Context.html). Most of the methods in MiniSSL::Context operate on the query value, and most of the logic is also in that class.  Some is in MiniSSL::ContextBuilder, it would probably be a good idea to refactor a bit.

Thoughts?

Reply:
I think at this point, let's not add too many blockers to getting 6.0 out. Do what you think is best 👍

Reply:
Sorry.  I know this is a bit of a PITA, but the ssl bind strings have seen a large increase in size with all the additional SSL options.  At present, the `ssl_bind` call in a config file is just taking simple objects (numbers, strings, boolean) as values in its hash, so currently it looks like:
```ruby
  ssl_bind '#{HOST}', '#{bind_port}', {
    cert: cert,
    key:  key,
    ca: ca,
    verify_mode: 'none',
    reuse: 'dflt'
  }
```
We could change it to the following, where the `size` and `timeout` keys are optional?
```ruby
  ssl_bind '#{HOST}', '#{bind_port}', {
    cert: cert,
    key:  key,
    ca: ca,
    verify_mode: 'none',
    reuse: {use: true, size: 2_000, timeout: 400}
  }
```

What do you think of that?  We still need to encode it to a string and then unpack it to create the SSL context.

Reply:
@nateberkopec I didn't quite use the style shown above, but something similar to what you were looking for.  Changed opening comment.